chore(many): remove hash symbol from internal links

ToMESSKa opened the pull request:
INSTUI-4750

**ISSUE:**
- remove # from InstUI documentation links (e.g.: https://instructure.design/#Alert) to smoothen AI agent consumption of documentation links

**TEST PLAN:**
- open the following links:
  https://instructure.design/pr-preview/pr-2167/Alert - it should open without error
  https://instructure.design/pr-preview/pr-2167/#Alert - it should open without error
  https://instructure.design/pr-preview/pr-2167/#Alert/#Guidelines - it should open without error and scroll to Guidelines
  https://instructure.design/pr-preview/pr-2167/Alert#Guidelines it should open without error and scroll to Guidelines
- on the preview page, open as many menu item from the navigation bar as possible, they should open without error and without reloading the whole page
- try to open the Themes and Icons menu too, they should open without error and without reloading the whole page
- try internal links e.g.: clicking on the findDOMNode link in the Accessing the DOM page, it should open without error and without reloading the whole page
- try external links e.g.: 'specific requirements' Focus management , they should open without error
- try the links and buttons on the home page (Changelog, Developer Quick Start, MIT etc), they should open without error and without reloading the whole page when the Nav is open
- try the version selector in Nav, it should work without error
- clicking on the InstUI logo in the Nav should lead to the main page
- try the links in the Table of Contents, they should work without error
- run it locally and open the following links e.g.:
  http://localhost:9090/Alert - it should open without error
  http://localhost:9090/#Alert - it should open without error
  http://localhost:9090/#Alert/#Guidelines - it should open without error and scroll to Guidelines
  http://localhost:9090/Alert#Guidelines it should open without error and scroll to Guidelines
- try every test case listed for the pr- preview in the local build too
- **note:** a 404.html is added to handle links here: https://github.com/instructure/instructure-ui/tree/gh-pages

ToMESSKa commented on packages/__docs__/webpack.config.mjs:
```diff
@@ -59,6 +59,7 @@ const config = merge(baseConfig, {
       directory: outputPath,
     },
     host: '0.0.0.0',
+    historyApiFallback: true,
```
this need to be able to refresh the page in dev build

matyasf reviewed (changes requested):
- We will need to rewrite internal links like the ones in `/accessing-the-dom` from `#bla` to `/bla`
- the internal anchor links like the ones in table of contents do not work

ToMESSKa commented:
> https://instructure.design/pr-preview/pr-2167/

@matyasf

- I fixed the links in the table of contents
- I removed the # from the links
- I also fixed a few other things so I extended the test plan
